src/CabalHelper/Compiletime/Compat/Environment.hs

```diff
@@ -1,6 +1,18 @@
+{-# LANGUAGE CPP #-}
 module CabalHelper.Compiletime.Compat.Environment where
 
-import System.Environment
+import qualified System.Environment (getEnvironment, setEnv)
+#ifndef mingw32_HOST_OS
+import qualified System.Posix.Env (setEnv)
+#endif
 
 lookupEnv :: String -> IO (Maybe String)
-lookupEnv var = do env <- getEnvironment; return (lookup var env)
+lookupEnv var = do env <- System.Environment.getEnvironment
+                   return (lookup var env)
```

Comment:
Unrelated style change, please revert. Don't lump such things in with other changes unless this was causing a build error on windows ;)

Reply:
It's was longer than 80 columns in one line. You want it to be one line or just in different commits?

Reply:
Oh I read over that `S.E.` qualifier, never mind that then. Do add a line break after the `do` though.